analyzers/src/SonarAnalyzer.CSharp/Extensions/AssignmentExpressionSyntaxExtensions.cs

```diff
@@ -0,0 +1,82 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using StyleCop.Analyzers.Lightup;
+
+namespace SonarAnalyzer.Extensions
+{
+    internal static class AssignmentExpressionSyntaxExtensions
+    {
+        public readonly record struct AssignmentMapping(ExpressionSyntax Left, ExpressionSyntax Right);
```

Comment:
SA1313 is raised for `record struct`. I checked the implementation and record struct support was added [7 months ago](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blame/feea263906c6ac13b832dbde2c2d15901795b6fb/StyleCop.Analyzers/StyleCop.Analyzers/NamingRules/SA1313ParameterNamesMustBeginWithLowerCaseLetter.cs#L70-L71):

I can suppress the FP in SQ or we can update the StyleCop dependency.

Reply:
> Open question is, whether it should be a nested type in AssignmentExpressionSyntaxExtensions or not.

If it's a type used only in this class as a helper, IMO is a good idea to be as it is now.  If we see the need in the future we might create a more generic type. @csaba-sagi-sonarsource  WDYT?

Reply:
> I can suppress the FP in SQ or we can update the StyleCop dependency.

@pavel-mikula-sonarsource Can we update StyleCop? Is something stopping us?

Reply:
If you mean updating the analyzer package reference that we use to analyze code, then sure, why not:
https://github.com/SonarSource/sonar-dotnet/blob/ca431303d8ebc86664f601901dbc6a7c5942f495/analyzers/StyleCopAnalyzers.targets#L4

If you mean updating ShimLayer, then we can, but it's not an easy task.

Reply:
I updated the StyleCop.Analyzers in eca89ecc. @pavel-mikula-sonarsource If this should be done in its own PR. Just drop me a line and I revert the change.

Reply:
I reverted the change and created #5769 instead.

Reply:
I agree to make it separate, with a clear PR title / commit message.

Reply:
Regarding the placement of the `AssignmentMapping` record sturct. I would leave it here for now. In case any other extension reuses it in the future, it will be moved to its own file.

Reply:
You can declare it like this to make it language-neutral. I don't see practical benefit in liming the type to expression. One day, we might even use it in the language facade.
```
public readonly record struct AssignmentMapping(SyntaxNode Left, SyntaxNode Right);
```